lib/reek/smells/unneeded_suppression.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Reek
+  module Smells
+    #
+    # An unneeded suppression is a Reek's comment disabling a unexistent smell.
+    #
+    # See {file:docs/Unneeded-Suppression.md} for details.
+    class UnneededSuppression
+      def initialize(context, smell_detector)
+        @context = context
+        @smell_detector = smell_detector
+      end
+
+      def sniff
+        if @smell_detector.sniff(@context).empty?
+          exp = @context.exp
+          ctx_source = exp.loc.expression.source_buffer.name
+          [SmellWarning.new(self,
```

Comment:
The reason seems to be that I suggested not deriving this class from SmellDetector because I had a different architecture in mind. Sorry for the confusion.